crates/swc_plugin_runner/src/cache.rs

```diff
@@ -0,0 +1,153 @@
+use std::{
+    env::current_dir,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{Context, Error};
+use parking_lot::Mutex;
+use swc_common::{
+    collections::AHashMap,
+    sync::{Lazy, OnceCell},
+};
+use wasmer::{Module, Store};
+use wasmer_cache::{Cache as WasmerCache, FileSystemCache, Hash};
+
+/// Version for bytecode cache stored in local filesystem.
+///
+/// This MUST be updated when bump up wasmer.
+///
+/// Bytecode cache generated via wasmer is generally portable,
+/// however it is not gauranteed to be compatible across wasmer's
+/// internal changes.
+/// https://github.com/wasmerio/wasmer/issues/2781
+const MODULE_SERIALIZATION_VERSION: &str = "v1";
+
+/// A shared instance to plugin's module bytecode cache.
+/// TODO: it is unclear how we'll support plugin itself in wasm target of
+/// swc, as well as cache.
+pub static PLUGIN_MODULE_CACHE: Lazy<PluginModuleCache> = Lazy::new(Default::default);
```

Comment:
We use it in `Lazy`, so `PluginModuleCache` does not need to have `OnceCell` internally

Reply:
What I wanted to achieve was setting value once with dynamic value (cache_root_path for fs), which can be set via onceCell::get. 

Maybe I should change into `PLUGIN_MODULE_CACHE: OnceCell<PluginModuleCache>` then remove inner's OnceCell?

Reply:
Ah, I see.
I think it may result in an error if a user invoke `npx swc` from a different directory than project root.